app/code/core/Mage/Catalog/Block/Product/View.php

```diff
@@ -61,9 +61,12 @@ protected function _prepareLayout()
             }
             $description = $product->getMetaDescription();
             if ($description) {
-                $headBlock->setDescription(($description));
+                $headBlock->setDescription($description);
             } else {
-                $headBlock->setDescription(Mage::helper('core/string')->substr($product->getDescription(), 0, 255));
+                $description = $product->getDescription();
+                if ($description) {
+                    $headBlock->setDescription(Mage::helper('core/string')->substr($description, 0, 255));
+                }
```

Comment:
Need to `setDescription()` is empty:
```suggestion 
                    $headBlock->setDescription(Mage::helper('core/string')->substr($description, 0, 255));
               } else {
                    $headBlock->setDescription('');
                }
```

Just in case it's called like this later: https://github.com/OpenMage/magento-lts/blob/2dad20ca0b2345267d7c52090308f216e649e41a/app/design/frontend/base/default/template/page/html/head.phtml#L17-L17